tests/hostpath_quota_test.go

```diff
@@ -0,0 +1,463 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/container"
+	pvc "github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/persistentvolumeclaim"
+	"github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/pod"
+	"github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/core/v1/volume"
+	sc "github.com/openebs/dynamic-localpv-provisioner/pkg/kubernetes/api/storage/v1/storageclass"
+	"github.com/openebs/dynamic-localpv-provisioner/tests/disk"
+)
+
+const (
+	// DiskImageSize is the default file size(1GB) used while creating backing image
+	DiskImageSize = 1073741824
+)
+
+var _ = Describe("TEST HOSTPATH XFS QUOTA LOCAL PV WITH NON-XFS FILESYSTEM", func() {
+	var (
+		pvcObj        *corev1.PersistentVolumeClaim
+		scObj         *storagev1.StorageClass
+		accessModes   = []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce}
+		capacity      = "5M"
+		podName       = "busybox-hostpath"
+		label         = "demo=hostpath-pod"
+		createdPvc    *corev1.PersistentVolumeClaim
+		pvcName       = "pvc-hp"
+		scNamePrefix  = "sc-hp-xfs"
+		scName        string
+		podObj        *corev1.Pod
+		labelselector = map[string]string{
+			"demo": "hostpath-pod",
+		}
+	)
+
+	When("StorageClass with valid xfs quota parameters is created", func() {
+		It("should create a StorageClass", func() {
+			By("building a StorageClass")
+			scObj, err = sc.NewStorageClass(
+				sc.WithGenerateName(scNamePrefix),
+				sc.WithLabels(map[string]string{
+					"openebs.io/test-sc": "true",
+				}),
+				sc.WithLocalPV(),
+				sc.WithHostpath("/var/openebs/integration-test"),
+				sc.WithVolumeBindingMode("WaitForFirstConsumer"),
+				sc.WithReclaimPolicy("Delete"),
+				sc.WithParameters(map[string]string{
+					"enableXfsQuota": "true",
+					"softLimitGrace": "20%",
+					"hardLimitGrace": "50%",
+				}),
+			)
+			Expect(err).To(
+				BeNil(),
+				"while building StorageClass with name prefix {%s}",
+				scNamePrefix,
+			)
+
+			By("creating StorageClass API resource")
+			scObj, err = ops.SCClient.Create(context.TODO(), scObj)
+			Expect(err).To(
+				BeNil(),
+				"while creating StorageClass with name prefix {%s}",
+				scNamePrefix,
+			)
+			scName = scObj.ObjectMeta.Name
+		})
+	})
+
+	When("pvc with storageclass "+scName+" is created", func() {
+		It("should create a pvc", func() {
+			By("building a PVC with StorageClass " + scName)
```

Comment:
`scName` is being set by the first `When` container block. Better to verify if it is empty or not. Add this to other blocks also wherever we are using values from the different blocks. You can use
```golang
Expect(scName).NotTo(BeEmpty(), "SC name should not be empty")
```